### PR TITLE
Add misspelled words from EC codebase

### PR DESCRIPTION
**Pull request by thughes**
See https://crrev.com/c/1678246 and https://crrev.com/c/1636848.

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -6891,6 +6894,7 @@ delagates->delegates
 delaloc->delalloc
 delalyed->delayed
 delapidated->dilapidated
+delaraction->declaration
```
Comment (luzpaz) [inline codespell_lib/data/dictionary.txt]: please add an entry for the plural  
`delaractions->declarations`

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -10124,6 +10128,7 @@ financialy->financially
 finanize->finalize
 finanlize->finalize
 fincally->finally
+finctionality->functionality
```
Comment (luzpaz) [inline codespell_lib/data/dictionary.txt]: please add these entries as well:
```
finctional->functional  
finctionalities->functionalities
```

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -20192,6 +20197,7 @@ sesssions->sessions
 setgit->setgid
 seting->setting
 setings->settings
+setion->section
```
Comment (luzpaz) [inline codespell_lib/data/dictionary.txt]: add plural entry as well

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt]: IIRC I requested you also add: `finctional->functional` :point_up_2:

**Comment by luzpaz** [comment]
Can't 'unresolve' previous comment that's been resolved, so I'll make a new one. Please add `finctional->functional` per previous request

**Comment by thughes** [comment]
Sorry, missed that.

**Comment by luzpaz** [comment]
Travis failed: https://travis-ci.org/codespell-project/codespell/jobs/622942851#L421-L425
(dictionary needs sorting)

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -4222,6 +4223,7 @@ cleaer->clear, clearer,
 cleaered->cleared
 cleaing->cleaning
 cleancacne->cleancache
+cleand->cleaned
```
Comment (arm-in) [inline codespell_lib/data/dictionary.txt]: also "cleans, clean"

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -10124,6 +10129,9 @@ financialy->financially
 finanize->finalize
 finanlize->finalize
 fincally->finally
+finctional->functional
```
Comment (arm-in) [inline codespell_lib/data/dictionary.txt]: also "fictional"

**Comment by thughes** [comment]
Ping. I think all comments were resolved now.

**Pull request merged**